pkgs/development/interpreters/octave/wrapper.nix

```diff
@@ -0,0 +1,119 @@
+{ stdenv, octave, buildEnv
+, makeWrapper, texinfo
+, octavePackages
+, extraLibs ? []
+, extraOutputsToInstall ? []
+, postBuild ? ""
+, ignoreCollisions ? false
+}:
+
+# Create an octave executable that knows about additional packages
+let
+  env = let
+    paths = extraLibs ++ [ octave ];
+    octavePath = "${octave}";
+    octaveExecutable = "${placeholder "out"}/bin/octave-cli";
+  in (buildEnv {
+    name = "${octave.name}-env";
+
+    inherit paths;
+    inherit ignoreCollisions;
+    extraOutputsToInstall = [ "out" ] ++ extraOutputsToInstall;
+
+    # During "build" we must first unlink the /share symlink to octave's /share
+    # Then, we can re-symlink the all of octave/share, except for /share/octave
+    # in env/share/octave, re-symlink everything from octave/share/octave and then
+    # perform the pkg install.
+    postBuild = ''
+      if [ -L "$out/bin" ]; then
+         unlink $out/bin
+         mkdir -p "$out/bin"
+         cd "${octavePath}/bin"
+         for prg in *; do
+             if [ -x $prg ]; then
+                makeWrapper "${octavePath}/bin/$prg" "$out/bin/$prg" --set OCTAVE_SITE_INITFILE "$out/share/octave/site/m/startup/octaverc"
+             fi
+         done
+         cd $out
+      fi
+
+      # Remove symlinks to the input tarballs, they aren't needed.
+      for tarball in $out/*.tar.gz; do
+          unlink $tarball
+      done
+
+      if [ -L "$out/share" ]; then
+          unlink "$out/share"
+          mkdir -p "$out/share"
+      fi
+
+      for f in ${octavePath}/share/*; do
+          ln -s -t $out/share $f
+      done
+
+      if [ -L "$out/share/octave" ]; then
+         unlink "$out/share/octave"
```

Comment:
Why not `rm`? :)

Reply:
Mostly because all the other wrapper.nix-s (lua, python) use this same method. I am not sure of the benefit of this method compared to `rm`.

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/f3dcd0b42e5f220d93aee46d502e0cda7783e841